server.go

```diff
@@ -525,8 +527,13 @@ func (server *Server) ServeCodec(codec ServerCodec) {
 // ServeRequest is like ServeCodec but synchronously serves a single request.
 // It does not close the codec upon completion.
 func (server *Server) ServeRequest(codec ServerCodec) error {
+	return server.ServeRequestContext(context.Background(), codec)
+}
+
+// ServeRequestContext is like ServeRequest but uses the provided context.
```

Comment:
Can you rather mirror the documentation for ServeRequest and mention what happens when the context is cancelled.

Reply:
Done!